clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/spring/client/configuration/CredentialsProviderConfiguration.java

```diff
@@ -39,6 +40,33 @@ public class CredentialsProviderConfiguration {
   @ConditionalOnMissingBean
   public CredentialsProvider camundaClientCredentialsProvider(
       final CamundaClientProperties camundaClientProperties) {
+    return switch (camundaClientProperties.getMode()) {
+      case basic -> buildBasicAuthCredentialsProvider(camundaClientProperties);
+      case saas, selfManaged -> buildOAuthCredentialsProvider(camundaClientProperties);
```

Comment:
The case `null` is not covered here.

Reply:
There's no reason. If the mode return `null` the switch would throw a `NullPointerException`. But even then it is impossible to be `null` as the `CamundaClientPropertiesPostProcessor` will throw an `IllegalArgumentException` if it finds an invalid client mode.

Reply:
Actually the tests do fail in case of null. Not sure why it sometimes goes through the properties post processor and sometimes nowt 🤔 

Anyway I'll handle it.